### PR TITLE
feat: add a frontmatter key for sharing image on the social media

### PR DESCRIPTION
**Pull request by mamieorine**
Fixes https://github.com/GoogleChrome/developer.chrome.com/issues/4734, Fixes https://github.com/GoogleChrome/developer.chrome.com/issues/3124

Changes proposed in this pull request:

- Adds a new frontmatter key named `imagesharing` for sharing image on the social media

### CODE REVIEW

**Comment by mamieorine** [comment]
@devnook I understand that a goal of both issues [3124](https://github.com/GoogleChrome/developer.chrome.com/issues/3124) and [4734](https://github.com/GoogleChrome/developer.chrome.com/issues/4734)  is having a separate frontmatter key that allows to only set the sharing image for the entire post so I have linked both issues with this PR.

File: site/_includes/partials/meta.njk [hunk 1]
```diff
@@ -26,7 +26,7 @@
 <meta property="og:type" content="website" />
 <meta property="og:url" content="{{ currentUrl }}" />
 
-{% set socialImage = thumbnail or hero or site.defaultSocialImage %}
+{% set socialImage = sharingimage or thumbnail or hero or site.defaultSocialImage %}
```
Comment (devnook) [inline site/_includes/partials/meta.njk:29]: Please use camelcase or underscore in var names.

File: site/en/blog/_example/index.md [hunk 2]
```diff
@@ -51,6 +51,11 @@ hero: 'image/BrQidfK9jaQyIHwdw91aVpkPiib2/EnMzOm0mBytBA3AzlCG6.png'
 # Required if there is a hero or social image
 alt: >
   An alternative text description of your hero or social image.
```
Comment (devnook) [inline site/en/blog/_example/index.md:53]: Please add information about social images to the handbook too, next to hero images https://developer.chrome.com/docs/handbook/how-to/add-a-blog-post/#hero-images

**Review by devnook — CHANGES_REQUESTED** [review]
Reviewed

**Comment by mamieorine** [comment]
@devnook  I have updated this PR. Thank you 😊

File: site/en/blog/_example/index.md [hunk 3]
```diff
@@ -51,6 +51,11 @@ hero: 'image/BrQidfK9jaQyIHwdw91aVpkPiib2/EnMzOm0mBytBA3AzlCG6.png'
...
+
+# Optional
+# Image sharing on social media should be at least 828 x 416
+# https://developer.chrome.com/docs/handbook/how-to/add-media/
+# sharingImage: 'image/BrQidfK9jaQyIHwdw91aVpkPiib2/EnMzOm0mBytBA3AzlCG6.png'
```
Comment (devnook) [inline site/en/blog/_example/index.md:58]: Looks like yaml fields should be with underscored - sorry if I misled you before.

Comment (mamieorine) [reply site/en/blog/_example/index.md]: I have updated YAML frontmatter key from camelcase to underscored

File: site/en/docs/handbook/how-to/add-a-blog-post/index.md [hunk 4]
```diff
@@ -37,6 +37,19 @@ shortcode snippet, it should look like this:
 
 Paste that into the `hero` field in your YAML frontmatter.
 
+### Social sharing image 
+
+Social sharing image is the visual thumbnail that appears when a post or page 
+is shared on social media platforms like Twitter and Facebook.
+
+Follow the [add media guide](/docs/handbook/how-to/add-media/) to upload your
+image to our CDN. Once you've uploaded the image, copy the path out of the
```
Comment (matthiasrohmer) [inline site/en/docs/handbook/how-to/add-a-blog-post/index.md:46]: I think as https://github.com/GoogleChromeLabs/chrome-gcs-uploader/pull/54#pullrequestreview-1299492930 is almost there, let's add a sentence here like "or use the sharing image generator at ..."

**Review by matthiasrohmer — APPROVED** [review]
Only one small request.

File: site/_includes/partials/meta.njk (hunk 1 above)
Comment (mamieorine) [reply site/_includes/partials/meta.njk]: I updated this

File: site/en/blog/_example/index.md (hunk 2 above)
Comment (mamieorine) [reply site/en/blog/_example/index.md:53]: I added the information about social images here: https://github.com/GoogleChrome/developer.chrome.com/pull/4981/files#diff-884a22587fc7726d735ce53ba4ac18e41f40ee70593213c811e573dfa3eaf428

**Comment by mamieorine** [comment]
@rachelandrew, forwarding to you for handbook updates about the social sharing image. Thank you 😊

File: site/en/docs/handbook/how-to/add-a-blog-post/index.md [hunk 5]
```diff
@@ -37,6 +37,22 @@ shortcode snippet, it should look like this:
 
 Paste that into the `hero` field in your YAML frontmatter.
 
+### Social sharing image 
+
+Social sharing image is the visual thumbnail that appears when a post or page 
+is shared on social media platforms like Twitter and Facebook.
+
+Follow the [add media guide](/docs/handbook/how-to/add-media/) to upload your
+image to our CDN or use the [sharing image generator](https://web-dev-uploads.web.app/sharing-image-generator)
```
Comment (rachelandrew) [inline site/en/docs/handbook/how-to/add-a-blog-post/index.md:46]: I get a 404 at that URL.

Comment (mamieorine) [reply site/en/docs/handbook/how-to/add-a-blog-post/index.md:46]: @rachelandrew  Sorry, Matthias and me have fixed this issue. The sharing image generator URL (https://web-dev-uploads.web.app/sharing-image-generator) is now available. Please re-review it again, thank you ☺️

**Review by rachelandrew — APPROVED** [review]

**Review by matthiasrohmer — APPROVED** [review]

**Pull request merged**